src/python/WMComponent/JobAccountant/JobAccountantPoller.py

```diff
@@ -70,43 +63,31 @@ def algorithm(self, parameters = None):
         logging.info("Found %d completed jobs" % len(completeJobs))
 
         if len(completeJobs) == 0:
-            # Then we have no work to do.  Bye!
             logging.debug("No work to do; exiting")
             return
 
-        while len(completeJobs) > self.accountantWorkSize:
+        while len(completeJobs) > 0:
             try:
                 jobsSlice = completeJobs[:self.accountantWorkSize]
                 completeJobs = completeJobs[self.accountantWorkSize:]
                 self.accountantWorker(jobsSlice)
+                if len(completeJobs) <= 100:
+                    self.accountantWorker(completeJobs)
+                    break
```

Comment:
I think these three lines just need to be removed. (from 74 to 76). Considering initial completeJobs = 90